game/game.go

```diff
@@ -708,3 +709,18 @@ func (g *Game) playActivatedAbility() {
 	g.Print()
 	panic("playActivatedAbility failed")
 }
+
+// Returns a binary representation of the game, json for now
+func (g *Game) serialized() []byte {
+	gameJson, _ := json.Marshal(g)
+	return gameJson
+}
+
+// Reset the game pointer for game.players, and return a deserialized game
```

Comment:
All the Permanents also have a game pointer that I think you will need to set as well, otherwise things will get borked.

It might help to have a "validate" function on the `Game` object that does things like loop through and check that all the self-pointers are accurate, just to catch problems like this.